### PR TITLE
SYS-2779 rename `round` to `era`

### PR DESCRIPTION
**Pull request by nahuseyoum**
This PR is a pure rename of `round` to `era` with no additional functionality. This rename is useful because it matches the terminology used by substrate's staking pallet and our offchain products.

### CODE REVIEW

**Review by fluorostani — APPROVED** [review]

File: pallets/parachain-staking/src/nomination_requests.rs [hunk 1]
```diff
@@ -47,18 +47,18 @@ impl<Balance: Copy> NominationAction<Balance> {
 }
 
 /// Represents a scheduled request that define a [NominationAction]. The request is executable
-/// iff the provided [RoundIndex] is achieved.
+/// iff the provided [EraIndex] is achieved.
```
Comment (thadouk) [inline pallets/parachain-staking/src/nomination_requests.rs:50]: existing typo?
```suggestion
/// if the provided [EraIndex] is achieved.
```

**Review by thadouk — APPROVED** [review]
Does this Era match the Substrate Era in terms of time?
Meaning is 1 Era in the staking pallet starts and end in an era of the framework & the sessions?
If not this could create some confusion.

**Comment by nahuseyoum** [comment]
> Does this Era match the Substrate Era in terms of time? Meaning is 1 Era in the staking pallet starts and end in an era of the framework & the sessions? If not this could create some confusion.

We don't use substrate's staking pallet on our parachain so there shouldn't be any overlap there. In terms of defining how long an era is, its configurable.

File: pallets/parachain-staking/src/nomination_requests.rs (hunk 1 above)
Comment (nahuseyoum) [reply pallets/parachain-staking/src/nomination_requests.rs:50]: Its not a typo, it means "if and only if"

Comment (alexpArtos) [reply pallets/parachain-staking/src/nomination_requests.rs:50]: It's a standard in maths books

Comment (alexpArtos) [reply pallets/parachain-staking/src/nomination_requests.rs:50]: But I do agree that it is ambiguous for anyone who does not have that background, and that we should be explicit and avoid abbreviations where possible

**Review by alexpArtos — APPROVED** [review]

**Review by alexpArtos — APPROVED** [review]

**Review by thadouk — APPROVED** [review]

**Review by fluorostani — APPROVED** [review]

**Pull request merged**